examples/basics/demo_hf_dataset.py

```diff
@@ -0,0 +1,103 @@
+"""
+Using huggingface dataset
+====================================================================================================
+
+This simple example shows how to load and prepare properly a huggingface dataset.
+Context: having a quick access to several datasets available under the huggingface format.
+Available datasets: https://huggingface.co/datasets?search=deepinv
+
+Here we use drunet_dataset (https://github.com/samuro95/GSPnP)
+"""
+
+# %%
+# Load libraries
+# ----------------------------------------------------------------------------------------
+#
+
+from datasets import load_dataset
+from torch.utils.data import IterableDataset, DataLoader
+from torchvision import transforms
+
+import deepinv as dinv
+
+
+# %%
+# Stream data from Internet
+# ----------------------------------------------------------------------------------------
+#
+
```

Comment:
Remove this newline to render the comment as plaintext. Then make it more readable: 

> Stream data from HuggingFace. Only a limited number of samples should be in memory and nothing should be saved on disk. The dataset is downloaded from https://huggingface.co/datasets/deepinv/drunet_dataset. ``type : datasets.iterable_dataset.IterableDataset``

<img width="412" alt="Screenshot 2024-07-21 235950" src="https://github.com/user-attachments/assets/c4ced3cf-7fe0-4ec4-a9e9-b5940a988e5b">